### ISSUE TITLE
Call to backward() obstructs script from terminating (maybe related to Windows 7)

### ISSUE THREAD

**Issue by Tobias314** [pytorch/pytorch#29383]
Whenever backward() is called to compute gradients the python script never terminates. The backward function is not per se blocking and all lines after it are still executed however the script just does not terminate. This may be related with the operation system which is Windows 7 or with the use of  conda. At least there is another [(post on Stack Overflow)](https://stackoverflow.com/questions/57857690/pytorch-autograd-obstructs-script-from-terminating)  describing the same problem in the same environment.

## To Reproduce

1. install PyTorch on windows 7 via conda (`conda create --name grad_test pytorch -c pytorch`)
1. run example code given below taken from the post on Stack Overflow

```
import torch

x = torch.randn(3, requires_grad=True)
y = x * 2
print(y)

gradients = torch.tensor([0.1, 1.0, 0.0001], dtype=torch.float)
y.backward(gradients)
print(x.grad)

print("all done")
```

## Expected behavior
Script runs and exits after printing "all done".

## Environment
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 7 Enterprise
CMake version: version 3.9.6

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1080
Nvidia driver version: 441.12
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.17.3
[pip] torch==1.3.0
[pip] torchvision==0.4.1
[conda] blas                      1.0                         mkl    anaconda
[conda] libblas                   3.8.0                    14_mkl    conda-forge
[conda] libcblas                  3.8.0                    14_mkl    conda-forge
[conda] liblapack                 3.8.0                    14_mkl    conda-forge
[conda] mkl                       2019.4                      245
[conda] mkl-service               2.3.0            py37hfa6e2cd_0    conda-forge
[conda] pytorch                   1.3.0           py3.7_cuda101_cudnn7_0    pytorch
[conda] torchvision               0.4.1                py37_cu101    pytorch

cc @ezyang @SsnL @albanD @zou3519 @gqchen @peterjc123

**Comment by peterjc123** [comment]
Well, I could not reproduce it with PyTorch 1.3.1 + Python 3.6 + CUDA 9.2 + Windows 10.

**Comment by albanD** [comment]
Can't reproduce either on linux / mac machines.

**Comment by peterjc123** [comment]
Cannot reproduce it with PyTorch 1.3.1 + Python 3.7 + CUDA 10.1 + Windows 10 + Conda too.

**Comment by irmo322** [comment]
I have the exact same problem on my laptop.

I'm also on windows 7 (x64), premium home edition.
I don't have any NVidia GPU on my laptop.
I installed python and pytorch via conda (python 3.7.3, pytorch 1.3.1).

I can post more info if needed.

**Comment by peterjc123** [comment]
Looks like it's a Windows-7 only problem. Sorry, I don't any PC with Windows 7 at hand. Would you please post which function in C++ it gets stuck through debugging with VS?

**Comment by irmo322** [comment]
Sorry but I use pytorch only via python and conda, I never compiled pytorch myself.
I don't feel confident to compile it and debug it with VS.

**Comment by VitaMusic** [comment]
I confirm the same problem with pytorch (python 3.6.9, pytorch 1.3.1) on windows 7 x64

**Comment by pemryan** [comment]
same problem with python 3.7.6, pytorch 1.5.1 on windows 7 x64

**Comment by pemryan** [comment]
occurs regardless of CUDA version. same phenomenon for 10.0, 10.11 10.2.

**Comment by RundownRhino** [comment]
Same problem on Python 3.7.0, PyTorch 1.5.1, CUDA 10.2. Windows7 x64, PyCharm.

Minimal code to reproduce the bug is:

```
x = torch.tensor(3., requires_grad=True)
y = x*2
y.backward()
```

**Comment by tomzx** [comment]
The problem is still present on the CPU only version of PyTorch 1.6.0, Python 3.7.5, Windows 7 x64.

Execution stack of the thread that is left alive:

```
0, ntoskrnl.exe!_misaligned_access+0x17e7
1, ntoskrnl.exe!KeWaitForMultipleObjects+0xf5d
2, ntoskrnl.exe!KeWaitForSingleObject+0x19f
3, ntoskrnl.exe!ExQueryAttributeInformation+0x72a
4, ntoskrnl.exe!KeSynchronizeExecution+0x3a23
5, ntdll.dll!ZwReleaseKeyedEvent+0xa
6, ntdll.dll!RtlTryAcquireSRWLockShared+0x1a0
7, ntdll.dll!RtlWakeConditionVariable+0x47
8, msvcp140.dll!Cnd_signal+0x11
9, torch_cpu.dll!torch::autograd::Engine::~Engine+0x1dc
10, ucrtbase.dll!iswctype+0x2b7
11, ucrtbase.dll!register_onexit_function+0x87
12, ucrtbase.dll!execute_onexit_table+0x31
13, torch_python.dll!torch::FunctionSignature::toString+0x226125
14, torch_python.dll!torch::FunctionSignature::toString+0x22624a
15, ntdll.dll!LdrShutdownProcess+0x1d1
16, ntdll.dll!RtlExitUserProcess+0x90
17, ucrtbase.dll!wassert+0x356
18, ucrtbase.dll!wassert+0x2ae
19, python37.dll!Py_Exit+0x1f
20, python37.dll!Py_SymtableStringObject+0x666
21, python37.dll!PyErr_PrintEx+0x50
22, python37.dll!PyRun_InteractiveLoopFlags+0x16f
23, python37.dll!PyRun_AnyFileExFlags+0x45
24, python37.dll!Py_UnixMain+0x50b
25, python37.dll!Py_UnixMain+0x5b3
26, python37.dll!PyErr_NoMemory+0x1a41c
27, python37.dll!Py_Main+0x77
28, python37.dll!Py_Main+0x52
29, python.exe+0x1258
30, kernel32.dll!BaseThreadInitThunk+0xd
31, ntdll.dll!RtlUserThreadStart+0x21
```

**Comment by peterjc123** [comment]
@tomzx Does the hotfix mentioned in https://support.microsoft.com/zh-cn/help/2582203/a-process-that-is-being-terminated-stops-responding-in-windows-7-or-in help?

**Comment by MingyuanZ-USTB** [comment]
I have the same question with python 3.6.4, win6 x64, torch 1.6.0 , cuda 10.1, do you find the solution yet?

**Comment by hudsontek** [comment]
Same issue here. I'm also using windows7 x64. Looks like it's a win7 only problem.

**Comment by pemryan** [comment]
> 
> 
> I have the same question with python 3.6.4, win6 x64, torch 1.6.0 , cuda 10.1, do you find the solution yet?

The best solution for now is to upgrade the OS to windows 10 or a Linux distro.

**Comment by gccollect** [comment]
> @tomzx Does the hotfix mentioned in https://support.microsoft.com/zh-cn/help/2582203/a-process-that-is-being-terminated-stops-responding-in-windows-7-or-in help?

Unfortunately it doesn't

**Comment by edmoren** [comment]
Also have this issue: 

Windows 7 x64 
python 3.8.6  (note that 3.9 is not compatible with Windows 7) 
torch 1.7.1+cpu 
torchaudio 0.7.2
torchvision 0.8.2+cpu 

Running in cpu only mode with no CUDA. 

backwards() causes script to never terminate

**Comment by jinkoo2** [comment]
Same here... pytorch 1.7.1 stable, python 3.7, anaconda3

os.kill(os.getpid(), 0) did not fix my issue.

**Comment by peterjc123** [comment]
This post is getting a bit stale. Please continue the discussion in https://github.com/pytorch/pytorch/issues/48888.
